lib/private/BackgroundJob/JobList.php

```diff
@@ -382,4 +383,23 @@ public function resetBackgroundJob(IJob $job): void {
 			->where($query->expr()->eq('id', $query->createNamedParameter($job->getId()), IQueryBuilder::PARAM_INT));
 		$query->executeStatement();
 	}
+
+	public function hasReservedJob(?string $className = null): bool {
+		$query = $this->connection->getQueryBuilder();
+		$query->select('*')
+			->from('jobs')
+			->where($query->expr()->neq('reserved_at', $query->createNamedParameter(0, IQueryBuilder::PARAM_INT)))
+			->setMaxResults(1);
+
+		if ($className !== null) {
+			$query->andWhere($query->expr()->eq('class', $query->createNamedParameter($className)));
+		}
+
+		try {
+			$result = $query->executeQuery();
+			return count($result->fetchAll()) > 0;
```

Comment:
we only load 1 entry, so you could also do:
```suggestion
			return $result->fetch() !== false;
```

Also I kind of die when `$result->closeCursor()` is not called.